execution/block-partitioner/src/v2/mod.rs

```diff
@@ -0,0 +1,719 @@
+// Copyright © Aptos Foundation
+
+#[cfg(test)]
```

Comment:
Still need to go through the logic in the module deeper but a high level, this module seems overloaded with doing too many things, can you think about refactoring/offloading some functionality to separate modules? For example, we have a `conflict_detector`and `dependency_edge_creator` modules in V1, that offloads some of the functionality. 

Also individual functions are pretty overloaded, let's think about breaking them into smaller pieces and add more comments.